wire.md

```diff
@@ -0,0 +1,1008 @@
+MUST NOT be included in responses to this request. Clients MUST be able to
+handle the hashes of unexpected types appearing in responses, and MAY choose
+for themselves whether to discard them or not.
+
+`future` MUST be set to either `1` or `0`.
+
+If `future = 1`, the responder SHOULD respond with future channel state changes
+as they become known to the responder, and the request SHOULD be held open
+indefinitely on both the requester and responder side until a Cancel Request is
+issued by the requester, or the responder elects to end the request by sending
+a Hash Response with `hash_count = 0`.
+
+If `future = 1` and a post that is part of the latest state for a channel is
+deleted, the responder MUST immediately send the hash of the next-latest
+piece of state of that same type as a Hash Response. For example, if the latest
+`post/topic` setting the channel's topic string is deleted by its author with a
+`post/delete` post, the hash of the second-latest `post/topic` for that channel
+SHOULD be sent.
+
+If `future = 0`, only the latest state posts will be included, and the request
+MUST NOT be held open.
+
+##### 6.3.2.6 Channel List Request
+
+Request a list of known channels from peers.
+
+field          | type               | desc
+---------------|--------------------|-----------------------------------
+`offset`       | `varint`           | number of channel names to skip (`0` to skip none)
+`limit`        | `varint`           | maximum number of channel names to return
+
+`msg_type` MUST be set to `6`.
+
+This request returns zero or more Channel List Response messages.
+
+If `limit` is 0, the responder MUST respond with all known channels (after
+skipping the first `offset` entries).
+
+The `offset` field can be combined with the `limit` field to allow clients to
+paginate through the list of all channel names known by a peer.
+
+#### 6.3.3 Responses
+Multiple responses MAY be generated for a single request, where results trickle
+in from the set of responding peers.
+
+Every response MUST begin with the message header detailed in Section 6.3.1,
+followed by bytes specific to the response `msg_type`, detailed in the sections
+that follow.
+
+Responses containing an unknown `req_id` SHOULD be ignored.
+
+Responders MUST set a response's `req_id` set to the same `req_id` of the
+request they are responding to.
+
+##### 6.3.3.1 Hash Response
+
+Respond with a list of zero or more hashes.
+
+field        | type                | desc
+-------------|---------------------|-------------------------------------
+`hash_count` | `varint`            | number of hashes to follow
+`hashes`     | `u8[hash_count*32]` | hashes, concatenated together
+
+`msg_type` MUST be set to `0`.
+
+A responder MUST send a Hash Response message with `hash_count = 0` to indicate
+that they do not intend to return any further hashes for the given `req_id` and
+they have concluded the request on their side.
```

Comment:
This makes responses that consist of zero hashes indistinguishable from there-might-be-data-but-I-wont-deliver-it.

Reply:
@AljoschaMeyer Is there a case where knowing the difference would be helpful?

Reply:
Fair =D I can't immediately come up with one.